cmd/apply/apply.go

```diff
@@ -0,0 +1,138 @@
+package apply
+
+import (
+	"os"
+	"path/filepath"
+
+	"github.com/konveyor/crane-lib/apply"
+	"github.com/konveyor/crane/internal/file"
+	"github.com/sirupsen/logrus"
+	"github.com/spf13/cobra"
+	"sigs.k8s.io/yaml"
+)
+
+type Options struct {
+	logger       logrus.FieldLogger
+	ExportDir    string
+	TransformDir string
+	OutputDir    string
+}
+
+func (o *Options) Complete(c *cobra.Command, args []string) error {
+	// TODO: @shawn-hurley
+	return nil
+}
+
+func (o *Options) Validate() error {
+	// TODO: @shawn-hurley
+	return nil
+}
+
+func (o *Options) Run() error {
+	return o.run()
+}
+
+func NewApplyCommand() *cobra.Command {
+	o := &Options{
+		logger: logrus.New(),
+	}
+	cmd := &cobra.Command{
+		Use:   "apply",
+		Short: "Apply the transformations to the exported resources and save results in an output directory",
+		RunE: func(c *cobra.Command, args []string) error {
+			if err := o.Complete(c, args); err != nil {
+				return err
+			}
+			if err := o.Validate(); err != nil {
+				return err
+			}
+			if err := o.Run(); err != nil {
+				return err
+			}
+
+			return nil
+		},
+	}
+
+	addFlagsForOptions(o, cmd)
+
+	return cmd
+}
+
+func addFlagsForOptions(o *Options, cmd *cobra.Command) {
+	cmd.Flags().StringVarP(&o.ExportDir, "export-dir", "e", "export", "The path where the kubernetes resources are saved")
+	cmd.Flags().StringVarP(&o.ExportDir, "transform-dir", "t", "transfrom", "The path where files that contain the transformations are saved")
+	cmd.Flags().StringVarP(&o.ExportDir, "output-dir", "o", "output", "The path where files are to be saved after transformation are applied")
+}
+
+func (o *Options) run() error {
+	// log := o.logger
+	a := apply.Applier{}
+
+	// Load all the resources from the export dir
+	p, err := filepath.Abs(o.ExportDir)
+	if err != nil {
+		// Handle errors better for users.
+		return err
+	}
+
+	t, err := filepath.Abs(o.TransformDir)
+	if err != nil {
+		return err
+	}
+
+	outputDir, err := filepath.Abs(o.OutputDir)
+	if err != nil {
+		return err
+	}
+
+	files, err := file.ReadFiles(p)
+	if err != nil {
+		return err
+	}
+
+	opts := file.PathOpts{
+		TransformDir: t,
+		ResourceDir:  p,
+		OutputDir:    outputDir,
+	}
+
+	for _, f := range files {
+		whPath := opts.GetWhiteOutFilePath(f.Info.Name())
+		_, statErr := os.Stat(whPath)
+		if !os.IsNotExist(statErr) {
+			o.logger.Debugf("resource file: %v is skipped due to white file: %v", f.Info.Name(), whPath)
+			continue
+		}
+
```

Comment:
nit: add a comment here each resource needs to have a transform or a whiteout

Reply:
It is expected that this be the case, probably should either error or handle that huh

Reply:
Added a comment for a to-do item to handle this.